File: sorting/bubble_sort.c
#include<stdio.h>

#define SIZE 10

void print_array(int a[], int size){
    for(int i = 0; i < size; i++){
        printf("%d ", a[i]);
    }
}

void bubble_sort(int a[], int size){
    int swap = 1, aux;
    while(swap){
        swap = 0;
        for(int i = 0; i < size-1; i++){
            if (a[i] > a[i+1]){
                aux = a[i];
                a[i] = a[i+1];
                a[i+1] = aux;
                swap = 1;
            }
        }
    }
}



int main(void){
    int array[] = {7, 5, 6,3, 2, 9, 5, 7, 6, 1};
    print_array(array, SIZE);
    printf("\n\n");
    bubble_sort(array, SIZE);
    print_array(array, SIZE);
}   

File: sorting/bubble_sort.py
from typing import List, Union
from random import randint

def bubble_sort(a: List[Union[int, float]]) -> None:
    swap = True
    while swap:
        swap = False
        for i in range(len(a) - 1):
            if a[i ] > a[i+1]:
                aux = a[i]
                a[i] = a[i+1]
                a[i+1] = aux
                swap = True 
    


if __name__ == '__main__':
    size_list = randint(15, 100)
    print(f"Size of the list: {size_list}")
    test_list = [randint(0, 100) for i in range(size_list)]

    print(f"Test list: {test_list}")
    bubble_sort(test_list)
    print(f"Sorted list: {test_list}")